### ISSUE TITLE
Additional alias names of custom emojis not accepted anymore

### ISSUE THREAD

**Issue by frdmn** [RocketChat/Rocket.Chat#18264]
### Description:

[According to our docs](https://docs.rocket.chat/guides/administrator-guides/custom-emoji), it should be possible to add multiple alias names to custom emojis, separated by space character: 

> Enter a name for your emoji, add space separated list of the aliases for the emoji which can be entered to display the emoji in the chats

This does not seem to be working anymore, the space character will be replaced by a dash instead.

### Steps to reproduce:

1. Add a new custom emoji
2. Give it multiple aliases, e.g. `first second`
3. Hit save, see how the alias gets transformed into: `first_second` 

### Expected behavior:

It should separate the alias input by space, as stated in the docs, and accept multiple alias names.

### Actual behavior:

Space character gets replaced by dash. Multiple aliases not possible.

### Server Setup Information:

- Version of Rocket.Chat Server: 3.4.0
- Operating System: Linux
- Deployment Method: Docker
- Number of Running Instances: Docker 
- DB Replicaset Oplog: Yes
- NodeJS Version: Docker
- MongoDB Version: Docker

##### Demo video

[<img width="796" alt="Bildschirmfoto 2020-07-15 um 10 45 26" src="https://user-images.githubusercontent.com/944459/87524368-6163be00-c688-11ea-964d-873f1d86ffd9.png">](https://up.frd.mn/0zWo1s8hFd.mp4)

**Comment by LeeThompson** [comment]
Seeing this too.

**Comment by LeeThompson** [comment]
This *appears* to be fixed as of 3.7.0.

**Comment by LeeThompson** [comment]
(This stat:stale thing because it hasn't been addressed or added to and then automatically closed is why these bugs stay.)

**Comment by RavuAlHemio** [comment]
I'm trying to resolve this in #23990; may I request a review?

**Comment by paulchen** [comment]
Today I investigated into several problems regarding custom emojis (cf. https://github.com/RocketChat/Rocket.Chat/issues/26979, https://github.com/RocketChat/Rocket.Chat/issues/26980).

In that process, I stumbled upon the problem described in this issue as well which still persists in Rocket.Chat 5.1.4.

**Comment by codewithmitesh** [comment]
Hey, @dudanogueira I want to work on the Issue, Can you assign it to me?

**Comment by RavuAlHemio** [comment]
I‘ve already proposed a fix in #23990; unfortunately, it appears the dev team has not yet come around to reviewing it…

**Comment by TafreedAhmad** [comment]
@dudanogueira Assign me if it's not solved yet, I've actually implemented it on my local machine.

![InkedScreenshot 2023-02-26 032229](https://user-images.githubusercontent.com/107949435/221382317-af891206-15eb-4674-906e-99615bd682fb.jpg)

**Comment by TafreedAhmad** [comment]
@dudanogueira @ggazzo 

![Screenshot 2023-02-26 032734](https://user-images.githubusercontent.com/107949435/221405964-b0ca7e66-0392-44d3-8a36-e6a97611c047.png)

https://github.com/RocketChat/Rocket.Chat/pull/28176

**Comment by LeeThompson** [comment]
I can't believe it's taken over 2 years...

**Comment by RavuAlHemio** [comment]
> I can't believe it's taken over 2 years...

Especially since there is more than one PR containing a fix...

**Comment by Shriyansh-20** [comment]
@paulchen @RavuAlHemio @frdmn @dudanogueira @LeeThompson Imterested, Please assign. Thanks.

**Comment by Curious-Goblin** [comment]
As per the docs, there is no mention which states of separating different alias by space, so this issue seems irrelevant. However the actual behavior is still there and seems relevant

**Comment by paulchen** [comment]
> As per the docs, there is no mention which states of separating different alias by space, so this issue seems irrelevant.

However, the field is still labelled "Aliases" indicating that multiple aliases are possible.

![Image](https://github.com/user-attachments/assets/a7bf77ee-addc-46da-a776-97cb2e3724e5)

**Comment by Curious-Goblin** [comment]
> However, the field is still labelled "Aliases" indicating that multiple aliases are possible.

yes, I think that the field seems to be for multiple alias names but they are a single name and there is also no mention of giving multiple alias names

**Comment by RavuAlHemio** [comment]
It's a bug. It was introduced when the list of aliases was passed through a "slugify" function to remove special characters -- unfortunately, that also replaced the spaces (which were the separator character for each alias) with underscores.

**Comment by frHimanshu** [comment]
Can i get assigned this issue?

**Comment by Satyam880880** [comment]
I want to work on this issues please assign me this issue

**Comment by paulchen** [comment]
> Can i get assigned this issue?

> I want to work on this issues please assign me this issue

The Rocket.Chat team does not assign GitHub issues to anyone who is not a team member. However, feel free to work on it and submit a PR when you think you fixed the issue. Please be aware that there are already multiple PRs for this issue.

By the way, next year this issue will celebrate its sixth birthday and eventually attend primary school.

**Comment by Anjali-yadav513** [comment]
Hi , I want to work on this issue . Here is my clear approach:
When a user removes an alias from a custom emoji, the update fails because the backend validation does not accept an empty alias list.
The UI sends either an empty string or an empty array, and the server rejects it.

The alias field is strictly validated. When it becomes empty([]), the server treats it as invalid and throws an error.


emoji.aliases = (emoji.aliases || []).filter(a => a.trim() !== ' ');

Users can remove aliases without errors the emoji update works smoothly.

Kindly asslign this issue to me . I would like to implement to the fix.

**Comment by paulchen** [comment]
Awesome, yet another PR that will be ignored by the Rocket.Chat team.

**Comment by Nandinibajaj28** [comment]
Hi, I’d like to work on this issue. Could you please assign it to me?

**Comment by RavuAlHemio** [comment]
Please read the previous comments. As @paulchen stated,

> The Rocket.Chat team does not assign GitHub issues to anyone who is not a team member.

The following is also relevant:

> Please be aware that there are already multiple PRs for this issue.

**Comment by Nandinibajaj28** [comment]
> Please read the previous comments. As [@paulchen](https://github.com/paulchen) stated,
> 
> > The Rocket.Chat team does not assign GitHub issues to anyone who is not a team member.
> 
> The following is also relevant:
> 
> > Please be aware that there are already multiple PRs for this issue.

Thanks for the clarification! I’ll look for another issue to contribute to.

**Comment by Yashika-code** [comment]
Hi, I’d like to work on this issue. Could you please assign it to me?

**Comment by RavuAlHemio** [comment]
I recommend that you read the previous comments as well. As @paulchen stated,

> The Rocket.Chat team does not assign GitHub issues to anyone who is not a team member.

and:

> Please be aware that there are already multiple PRs for this issue.

**Comment by wineerteam** [comment]
Hi team ,

I’ve gone through the issue description and comments.
The problem looks clear and reproducible on my side.

I’d like to work on this issue and submit a fix.
Please let me know if I can proceed.

Thanks!

**Comment by RavuAlHemio** [comment]
> I’ve gone through the issue description and comments.

I'll give you the benefit of the doubt and try to rephrase one of the previous statements in the comments of this issue:

> Please be aware that there are already multiple PRs for this issue.

This means that multiple people have already submitted fixes for this issue (via pull request) and the Rocket.Chat team has not acted upon them (yet?).

You can also try fixing the issue and opening your own pull request, but I doubt the Rocket.Chat team will treat it differently than all the previous ones.

**Comment by wineerteam** [comment]
Hi @RavuAlHemio,

Thank you for taking the time to explain the situation and for clarifying
the context around the existing PRs.

I understand that there are already multiple proposed fixes and that the
maintainers haven’t acted on them yet. I’ll take some time to review the
existing PRs more carefully and see if there’s any meaningful improvement
or different approach I can contribute.

Thanks again for the guidance — I appreciate it.

Best regards,
Sunil

**Comment by paulchen** [comment]
Hi @wineerteam,

> I understand that there are already multiple proposed fixes and that the
> maintainers haven’t acted on them yet. I’ll take some time to review the
> existing PRs more carefully and see if there’s any meaningful improvement
> or different approach I can contribute.

I think you're wasting your time here. The problem here is not the quality of the already existing PRs. The problem is that the Rocket.Chat team doesn't care about them and this GitHub issue as a whole. Hence, we don't even know whether any of the proposed fixes meets the standards the Rocket.Chat team aims for.

I think there are plenty of other issues or even other projects where your contribution would add more benefit.

**Comment by wineerteam** [comment]
Thanks for the feedback!

**Comment by Manohar-1369** [comment]
I reproduced this using the official Docker image `registry.rocket.chat/rocketchat/rocket.chat:latest` (pulled on 2026-01-08). The alias field still replaces space-separated values into a single alias instead of creating multiple aliases as documented.

There is a fork commit (RavuAlHemio: `9456a132940a45aec191ed2c8d77b55c364af3ca`) which changes emoji alias handling to first split on spaces/commas and then slugify each alias. That approach appears correct and solves the issue in principle, but it was never merged into the official repo.

I am planning to implement a proper fix in the official Rocket.Chat repository, including tests and documentation updates, and will open a PR referencing this work.

**Comment by RavuAlHemio** [comment]
Here's an update since my PR #23990 was recently closed: the issue has been at least partially fixed in #32991; multiple emoji aliases can be separated using commas and are then correctly interpreted.

Whether or not Rocket.Chat should also support multiple emoji aliases separated by spaces is a matter of taste; I don't really mind as long as there is _some_ way to specify multiple aliases.

**Comment by ggazzo** [comment]
the problem is actually with docs we use `,` to separate not ` ` (space)
